src/plugins/map/MapObject.cpp

```diff
@@ -1260,67 +1281,99 @@ void MapObjectMapLoc::UpdateText()
 
 //----------------------------------------------------------------------------
 
-MapObjectMapLoc* GetMapLocByIndex(size_t index)
+MapObjectMapLoc* MapLocTemplate::GetMapLoc()
 {
-	--index;
-
-	if (index >= sMapLocs.size())
-		return nullptr;
-
-	return sMapLocs[index];
+	return m_MapLoc;
 }
 
-MapObjectMapLoc* GetMapLocByTag(const std::string& tag)
+MapLocTemplate* GetMapLocByTag(const std::string& tag)
 {
-	auto iter = sLocationsMap.find(tag);
-	if (iter == sLocationsMap.end())
-		return nullptr;
+	for (auto maplocTemplate : gMapLocTemplates)
+	{
+		if (!strcmp(maplocTemplate->GetMapLoc()->GetTag().c_str(), tag.c_str()))
```

Comment:
You're comparing to std::strings here:
`if (maplocTemplate->GetMapLoc()->GetTag() == tag)`